modules/storage/minio.go

```diff
@@ -276,8 +276,10 @@ func (m *MinioStorage) Delete(path string) error {
 }
 
 // URL gets the redirect URL to a file. The presigned link is valid for 5 minutes.
-func (m *MinioStorage) URL(path, name string) (*url.URL, error) {
-	reqParams := make(url.Values)
+func (m *MinioStorage) URL(path, name string, reqParams url.Values) (*url.URL, error) {
+	if reqParams == nil {
+		reqParams = make(url.Values)
+	}
 	// TODO it may be good to embed images with 'inline' like ServeData does, but we don't want to have to read the file, do we?
 	reqParams.Set("response-content-disposition", "attachment; filename=\""+quoteEscaper.Replace(name)+"\"")
```

Comment:
It should make a clone, but not modify the `reqParams` directly.

Reply:
95072ab1e8d7ad2cf441ffcdab641bebd1a0bc11